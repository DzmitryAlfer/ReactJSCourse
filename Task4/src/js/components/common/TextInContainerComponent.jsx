import React from 'react';
import PropTypes from 'prop-types';
import {TextComponent} from "./TextComponent";

export const TextInContainerComponent = ({text, containerClass, textClass}) => (
    <div className={containerClass}>
        <TextComponent text={text} className={textClass} />
    </div>
);

TextInContainerComponent.propTypes = {
  text: PropTypes.string,
  containerClass: PropTypes.string,
  textClass: PropTypes.string,
};