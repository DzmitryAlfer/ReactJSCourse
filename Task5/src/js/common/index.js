export const SortBy = Object.freeze({
    RELEASE_DATE: 'release_date',
    RATING: 'vote_average',
});

export const SortOrder = Object.freeze({
    ASC: 'asc',
    DESC: 'desc',
});