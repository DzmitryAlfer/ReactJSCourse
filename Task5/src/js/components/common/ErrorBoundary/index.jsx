import React, { PureComponent } from 'react';
import * as actions from "../../../actions";
import {connect} from "react-redux";

class ErrorBoundary extends PureComponent {
    componentDidCatch(error, info) {
        this.props.dispatch(actions.app.setError(true));
    }

    render(){
        if(this.props.hasError) {
            return (<h1>Ooops, Something went wrong</h1>);
        }

        return this.props.children;
    }
}

const mapStateToProps = state => {
    const { appReducer } = state;

    return {
        hasError : appReducer.hasError
    }
};

export default connect (mapStateToProps)(ErrorBoundary);