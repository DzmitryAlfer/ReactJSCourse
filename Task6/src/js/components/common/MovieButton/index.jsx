import React from 'react';
import PropTypes from 'prop-types';

export const MovieButton = ({text, onButtonClick, containerClass, buttonClass}) => (
    <div className={containerClass}>
        <button onClick={onButtonClick} className={buttonClass}>{text}</button>
    </div>
);

MovieButton.defaultProps = {
    text: 'SEARCH',
    onButtonClick: () => {},
};

MovieButton.propTypes = {
    text: PropTypes.string.isRequired,
    containerClass: PropTypes.string,
    buttonClass: PropTypes.string,
    onButtonClick: PropTypes.func.isRequired,
};