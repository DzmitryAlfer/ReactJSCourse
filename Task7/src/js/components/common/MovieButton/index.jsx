import React from 'react';
import PropTypes from 'prop-types';
import {withRouter} from 'react-router-dom'

export const MovieButton = ({text, onButtonClick, containerClass, buttonClass}) => (
    <div className={containerClass}>
        <button onClick={onButtonClick} className={buttonClass}>{text}</button>
    </div>
);

export const LinkMovieButton = withRouter((props) => {
    return (<MovieButton {...props} onButtonClick={
        () => {props.history.push('/search');}
    }/>);
});

MovieButton.defaultProps = {
    text: 'SEARCH',
    onButtonClick: () => {},
};

MovieButton.propTypes = {
    text: PropTypes.string.isRequired,
    containerClass: PropTypes.string,
    buttonClass: PropTypes.string,
    onButtonClick: PropTypes.func.isRequired,
};