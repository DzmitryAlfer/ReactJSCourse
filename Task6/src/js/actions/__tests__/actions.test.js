import * as actions from "../";
import {SearchBy, SortBy} from "../../common";

describe('actions', () => {
    test('app.setError', ()=> {
        const action = actions.app.setError(true);
        expect(action).toMatchSnapshot();
    });

    test('movies.fetchMovies', ()=> {
        const action = actions.movies.fetchMovies('query');
        expect(action).toMatchSnapshot();
    });

    test('movies.fetchRelatedMovies', ()=> {
        const action = actions.movies.fetchRelatedMovies({id:1});
        expect(action).toMatchSnapshot();
    });

    test('movies.receiveMovies', ()=> {
        const action = actions.movies.receiveMovies([{id:1}, {id:2}]);
        expect(action).toMatchSnapshot();
    });

    test('movies.receiveRelatedMovies', ()=> {
        const action = actions.movies.receiveRelatedMovies([{id:1}, {id:2}]);
        expect(action).toMatchSnapshot();
    });

    test('movies.sortMovies', ()=> {
        const action = actions.movies.sortMovies(SortBy.RATING);
        expect(action).toMatchSnapshot();
    });

    test('movies.searchByMovies', ()=> {
        const action = actions.movies.searchByMovies(SearchBy.GENRES);
        expect(action).toMatchSnapshot();
    });
});