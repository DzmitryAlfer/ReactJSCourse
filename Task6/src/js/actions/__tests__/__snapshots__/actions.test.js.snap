// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions app.setError 1`] = `
Object {
  "hasError": true,
  "type": "SET_ERROR",
}
`;

exports[`actions movies.fetchMovies 1`] = `
Object {
  "searchQuery": "query",
  "type": "FETCH_MOVIES",
}
`;

exports[`actions movies.fetchRelatedMovies 1`] = `
Object {
  "movie": Object {
    "id": 1,
  },
  "type": "FETCH_RELATED_MOVIES",
}
`;

exports[`actions movies.receiveMovies 1`] = `
Object {
  "movies": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  "type": "RECEIVE_MOVIES",
}
`;

exports[`actions movies.receiveRelatedMovies 1`] = `
Object {
  "movies": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  "type": "RECEIVE_RELATED_MOVIES",
}
`;

exports[`actions movies.searchByMovies 1`] = `
Object {
  "searchBy": "genres",
  "type": "SEARCH_BY_MOVIES",
}
`;

exports[`actions movies.sortMovies 1`] = `
Object {
  "sortBy": "vote_average",
  "type": "SORT_MOVIES",
}
`;
